cql3/expr/prepare_expr.cc

```diff
@@ -1034,7 +1034,7 @@ test_assignment_function_call(const cql3::expr::function_call& fc, data_dictiona
 
 static assignment_testable::test_result expression_test_assignment(const data_type& expr_type,
                                                                    const column_specification& receiver) {
-    if (receiver.type->underlying_type() == expr_type->underlying_type()) {
+    if (receiver.type->underlying_type() == expr_type->underlying_type() || (receiver.type == long_type && expr_type->is_counter())) {
```

Comment:
I'm not sure that I understood you correctly, but are you saying that before this patch sum(counter) didn't work and after the patch, it does? Can we have a test for this case (and perhaps even an issue about it)?

Reply:
At this point in the series, this code isn't called with aggregate functions.